lib/resty/websocket/client.lua

```diff
@@ -256,7 +261,11 @@ function _M.connect(self, uri, opts)
         return nil, "bad HTTP response status line: " .. header
     end
 
-    return 1
+    if not keep_response then
+        header = nil
+    end
+
+    return 1, nil, header
```

Comment:
So regarding the early-exit with `:getreusedtimes()`, perhaps we do `return 1, nil, "connection reused"`? So `"connection reused"` is our token value? Since we will likely not enable upstream keepalive in lua-resty-websocket-proxy as discussed, we may not have to deal with it but at least the reused-socket case is handled. Whether this patch would be accepted upstream I have my doubts but we can revisit later.